File: 0205-isomorphic-strings/0205-isomorphic-strings.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        
        
        	int m = (int)t.size();
	string temp = "";
	map<char , int>mp1;
	map<char , char>mp2;
	for(int i = 0 ; i < m ; i ++){

		if(mp1[t[i]] != 1){
			mp1[t[i]] = 1;
			mp2[s[i]] = t[i];
		}
		else {
			continue;
		}

	}

	for(int i = 0 ; i < m ; i ++){
		s[i] = mp2[s[i]];
	}

	return s == t;
        
    }
};

File: 0205-isomorphic-strings/0205-isomorphic-strings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
            mapST, mapTS = {}, {}

            for e1, e2 in zip(s, t):

                if ((e1 in mapST and mapST[e1] != e2) or
                        (e2 in mapTS and mapTS[e2] != e1)):
                    return False
                mapST[e1] = e2
                mapTS[e2] = e1

            return True
